_aicspylibczi/ImageFactory.cpp

```diff
@@ -104,34 +128,37 @@ namespace pylibczi {
       shape.emplace_back(size.h);
       shape.emplace_back(size.w);
 
-      std::shared_ptr<Image> image = s_pixelToImage[pixelType](shape, pixelType, plane_coordinate_, box_, index_m_);
+      size_t mem_index = 0;
+
+      std::shared_ptr<Image> image = s_pixelToImage[pixelType](shape, pixelType, plane_coordinate_, box_,
+          m_imgContainer.get(), mem_index_, index_m_);
       if (image==nullptr)
           throw std::bad_alloc();
       image->loadImage(bitmap_ptr_, channels);
-
-      return std::shared_ptr<Image>(image);
+      auto sImage = std::shared_ptr<Image>(image);
```

Comment:
Just wondering, is this line needed?  shared_ptr should increment its ref on copy.

Reply:
I would have used a unique_ptr but you can't use the polymorphic trick with unique_ptrs whereas you can with shared pointers. The other thing is that I wanted to add it to the m_imgContainer member and return it to the caller in case they wanted to structure the images differently. 

Does that address your concern or am I missing something?

Reply:
The variable `image` is already a shared_ptr.  So you can just pass it around by value without creating the `sImage` variable.  That's what I was getting at.

Reply:
Thanks I totally didn't see that but yes your 100% right. thanks for catching that.